### DISCUSSION TITLE
avif

### DISCUSSION THREAD

**Discussion by amirreza-sf80** [codingjoe/django-pictures#164] [Ideas]
hello
thanks for you great work
since 
https://pypi.org/project/pillow-avif-plugin/
now exists
you think maybe we could have avif support as an optional installation?
like 
`pip install django-pictures[avif]` to install the plugin as well

**Comment by codingjoe** [comment]
Hi @amirreza8002,

Thanks for reaching out. Can you elaborate a bit more on why you'd prefer this, over just adding the plugin directly as a dependency? I am not averse to adding extra installs, but in this case, we don't have any core related to the package. Additionally, there are numerous Pillow extensions, making it harder to draw a line.

Curious to hear your thoughts.

Best!
Joe

**Comment by amirreza-sf80** [comment]
hi there @codingjoe 
my suggestion actually came from the feeling that you might want to keep the dependencies to a minimal amount and thought this might be a workaround.
but I don't have any reason to insist on it myself, my understanding is that avif is better suited for web (correct me if I'm wrong, I'm no expert), thus I have no problem with installing the dependency since I will be using it anyway.

with regards
amirreza

**Comment by amirreza-sf80** [comment]
hi @codingjoe 
sorry to un-bury this.

from a look around that i had, `models.py` and `tasks.py` ar the only paces that do formatting.

am i wrong on that?

i should be able to add the avif support, if it's something of interest for this package

**Reply by codingjoe** [reply]
Hi @amirreza8002,

No worries. I am always happy to help. It's not that easy. Django imports Pillow to check its dimensions. So this is a bigger undertaking, as you'd wish. I'm donating to Pillow and will wait for them to get it implemented, I'd invite you to do the same. I noticed that they have been very busy on this for the last couple of weeks, so I am hoping this will see the light of day soon.

Cheers!
Joe
